packages/blade/src/components/Tabs/_decisions/decisions.md

```diff
@@ -0,0 +1,385 @@
+    <button role="tab">tab item 1</button>
+    <button role="tab">tab item 2</button>
+    <button role="tab">tab item 3</button>
+  </div>
+  
+  <div>
+    <section role="tabpanel"></section>
+    <section role="tabpanel"></section>
+    <section role="tabpanel"></section>
+  </div>
+</div>
+```
+
+We need to achieve the above HTML structure, and to do this with the above API it will get very hacky since now we need to use React.cloneElement and save those JSX nodes in state and render them in appropriate place. 
+
+```html
+<Tabs>
+  // this will be the button with role="tab", need to React.cloneElement and move it to the right place, and while placing also need to make sure it has a parent of role="tablist"
+  <Tab value={} leading={} trailing={} title={}>  
+    // this is role="tabpanel" need to React.cloneElement and move it to the right place
+    {...the jsx content for the content of the tab} 
+  </Tab>
+</Tabs>
+```
+
+#### 2. The JSX doesn't map well with the visual hierarchy / the HTML structure
+
+If you notice the HTML structure pointed out in point #1 you'll see that it's 1:1 with the JSX API that i'm proposing. 
+
+The current API that is proposed also maps 1:1 with the visual hierarchy of the component, but the second one doesn't map very well.  
+
+![Frame 1 (11)](https://github.com/razorpay/blade/assets/35374649/9058d53f-aa5b-4a45-9111-5bfbdafa8f27)
+
+#### 3. Extensibility
+
+The current API is also much more easier to extend in case any future usecase arises, for example let's say we want to use these tabs like a segmented control (which works like radio buttons but doesn't associate with any "tabpanels" we can simply remove the tabpanels and change the underlying html structure to accommodate for any future usecases. 
+
+Check how reshaped is allowing this [extensibility](https://reshaped.so/content/docs/components/tabs#using-in-forms) - if you pass `name` prop it will convert the tabs to work like a Radio. 
+
+#### 4. API Familiarity 
+
+The current API that is proposed is also much more familiar and is similar to how many other design systems are approaching compared to the second one: 
+
+- [Reshaped](https://reshaped.so/content/docs/components/tabs)
+- [Chakra](https://chakra-ui.com/docs/components/tabs/usage)
+- [MUI](https://mui.com/material-ui/react-tabs/#experimental-api)
+- [Atlassian](https://atlassian.design/components/tabs/examples)
+- [RadixUI](https://www.radix-ui.com/primitives/docs/components/tabs)
+- [HeadlessUI](https://headlessui.com/react/tabs)
+
+
+## Motion
+
+Check the motion [documentation](https://www.figma.com/proto/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?type=design&node-id=4005-31894&t=b3OlPHxPuFH1BAIZ-0&scaling=min-zoom&page-id=3698%3A13772) here.
+
+## Accessibility
+
+Tabs will follow the [WAI-ARIA Tabs](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/) pattern.
+
+- [Keyboard navigation](https://www.w3.org/WAI/ARIA/apg/patterns/tabs/#keyboardinteraction)
+
+## Open Questions
```

Comment:
Are the tabs SEO-friendly by default, or do we need to make any adjustments?
Will the tab panels be server-rendered to facilitate SEO?

Reply:
By default all TabPanels be mounted on DOM and we will show/hide them via CSS to ensure SEO. 

And additionally, we will also introduce a `isLazy` prop which if present, will unmount the TabPanels from the DOM, could be useful if imagine inside your TabPanel you are doing API requests and don't want to mount/init them.